compiler/shared_ast/interpreter.ml

```diff
@@ -777,20 +750,21 @@ let rec evaluate_expr :
     match Mark.remove e with
     | ELit (LBool true) -> Mark.add m (ELit LUnit)
     | ELit (LBool false) ->
-      Message.error ~pos:(Expr.pos e') "Assertion failed:@\n%a"
+      Message.warning "Assertion failed:@ %a"
         (Print.UserFacing.expr lang)
         (partially_evaluate_expr_for_assertion_failure_message ctx lang
-           (Expr.skip_wrappers e'))
+           (Expr.skip_wrappers e'));
```

Comment:
You raise a warning and then you raise a runtime error ?

Reply:
Ah yes, this one is a bit tricky: here we indeed want to benefit from the greater powers we have in the interpreter to print a more detailed message, and arbitration was needed between that and the unification of error handling between runtime and interpreter.

The choice I made was to print the more detailed message as a side-effect, then raise the exception to be handled in the common path ; but indeed that will make the console show `[WARNING]` with the detail and then `[ERROR]` with the position and the message. The alternatives, either specific error handling of this exception at this point, or somehow embedding the added information into the runtime exception, didn't seem worth it.

Maybe this would be better if we add a non-raising variant for `Message.error` and use that here, though ?